src/AudioInterface.cpp

```diff
@@ -542,13 +546,32 @@ void AudioInterface::fromSampleToBitConversion
         tmp_8 = static_cast<int8_t>(tmp_sample);
         std::memcpy(output, &tmp_8, 1); // 8bits = 1 bytes
         break;
-    case BIT16 :
+    case BIT16 : {
         // 16bit integer between -32768 to 32767
         // original scaling: tmp_sample = floor( (*input) * 32768.0 ); // 2^15 = 32768.0
+        tmp_sample = double(*input);
+        if (fabs(tmp_sample) >= 1.0) {
+          mClipCount++;
+          mPeakMagnitude = std::max(mPeakMagnitude,fabs(tmp_sample));
+          if (mClipCount==mNextWarning) {
+            double mPeakMagnitudeDB = 20.0 * std::log10(mPeakMagnitude);
+            std::cerr << "*** AudioInterface.cpp: Audio HARD-CLIPPED on output to Internet!\n"
+                      << "\tReduce your input level(s) by " << mPeakMagnitudeDB << " dB.\n";
+            if (mClipCount>1) {
+              std::cerr << "\tMaximum amplitude over the last "
+                      << mNextWarning << " audio samples was "
+                        << mPeakMagnitude << ", or " << mPeakMagnitudeDB << " dBFS\n";
+            }
+            mPeakMagnitude = 0.0; // reset for next group measurement
+            if (mNextWarning < 1000) { // don't let it stop reporting for too long
+              mNextWarning *= 10;
+            }
```

Comment:
I'm missing where you bump mNextWarning or reset mClipCount

Reply:
I changed the algorithm at the last stroke from mNextWarning going up by factors of 10 to leveling off at 1000.  For that to retrigger every 1000 samples, we do need to reset mClipCount, which originally would keep counting.  However, to avoid the previous warnings at 1, 10, and 100, I have to reset to 101, which seems not nice.  I think I will rewrite this from scratch and see how it comes out.  I want to redo the PR from scratch anyway given the creep-in below from unknown sources that I cannot even see in my own version of the PR branch!